### DISCUSSION TITLE
Integrating IPFS?

### DISCUSSION THREAD

**Discussion by aronchick** [rook/rook#9474] [Q&A]
* Feature Request
Has anyone looked at integrating IPFS?

**What should the feature do:**
Ideally, there would be an IPFS provider, using the UnixFS feature, that would allow trivial read/write via UnixFS to the globally distributed filecoin network. (https://github.com/ipfs/go-unixfs)

**What is use case behind this feature:**
There are large and useful datasets on IPFS, and enabling native Kubernetes integration could be very useful for any teams needing to consume them.

**Comment by leseb** [comment]
I just learned about IPFS with this issue, I'm not sure how we would integrate with it. So essentially you have the IPFS network and then unixfs on top to essentially expose the data through a known filesystem interface, right?

I'm not sure what the proposal is for Rook, what kind of integration are talking about exactly?

**Reply by BlaineEXE** [reply]
> There are large and useful datasets on IPFS, and enabling native Kubernetes integration could be very useful for any teams needing to consume them.

From this statement, it seems that a CSI attacher driver is really the thing that would be most helpful, assuming the IPFS data is stored outside of Kubernetes.

I found an old IPFS project related to K8s, but it is archived now: There are large and useful datasets on IPFS, and enabling native Kubernetes integration could be very useful for any teams needing to consume them. https://github.com/ipfs-inactive/kubernetes-ipfs

It seems that it might be best to approach the IPFS community and see what their past efforts have been and current/future plans are. I only found a fairly old discussion here: https://discuss.ipfs.io/t/ipfs-as-a-storage-option-on-kubernetes/4506/3
